api/fees/data.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math"
+	"math/big"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/vechain/thor/v2/api/utils"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/cache"
+	"github.com/vechain/thor/v2/chain"
+)
+
+func newFeesData(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedSize uint32) *FeesData {
+	cacheSize := int(math.Min(float64(backtraceLimit), float64(fixedSize)))
+	maxBacktraceLimit := uint32(math.Max(float64(backtraceLimit), float64(cacheSize)))
+	return &FeesData{
+		repo:              repo,
+		cache:             cache.NewPrioCache(cacheSize),
+		bft:               bft,
+		cacheSize:         uint32(cacheSize),
+		maxBacktraceLimit: maxBacktraceLimit,
+	}
+}
+
+func newFeesCacheEntry(header *block.Header) *FeeCacheEntry {
+	baseFee := header.BaseFee()
+	if baseFee == nil {
+		baseFee = big.NewInt(0)
+	}
+	return &FeeCacheEntry{
+		baseFee:      (*hexutil.Big)(baseFee),
+		gasUsedRatio: float64(header.GasUsed()) / float64(header.GasLimit()),
+	}
+}
```

Comment:
Only one usage, we can leverage the complexity to a different method. Perhaps getBaseFee ?
```suggestion
func getBaseFee(baseFee *big.Int) *hexutil.Big {
	if baseFee != nil {
		return (*hexutil.Big)(baseFee)
	}
	return (*hexutil.Big)(big.NewInt(0))
}

```

Reply:
Done